File: app/src/main/java/com/example/olesya/boardgames/MenuFragment.java
package com.example.olesya.boardgames;

import android.content.pm.PackageManager;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.olesya.boardgames.databinding.FragmentMenuBinding;

public class MenuFragment extends Fragment {

    private FragmentMenuBinding mBinding;
    private MenuViewModel mvmodel;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
        mBinding = DataBindingUtil.inflate(inflater, R.layout.fragment_menu, container, false);
        mvmodel = new MenuViewModel(this);
        initListeners();
        ((AppCompatActivity) getActivity()).getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        return mBinding.getRoot();
    }

    private void initListeners() {
        mBinding.search.setOnClickListener(mvmodel.getOnSearchClickListener(getActivity(), this));
        mBinding.start.setOnClickListener(mvmodel.getOnStartClickListener(mBinding.switcher,
                mBinding.playerName, mBinding.playerNum, mBinding.ptsNum));
        mBinding.switcher.setOnCheckedChangeListener((buttonView, isChecked) -> {
            if (isChecked) {
                mBinding.statusMsg.setText("Server, waiting for connections");
                mBinding.search.setVisibility(View.GONE);
                mBinding.playerName.setVisibility(View.GONE);
                mBinding.screenContainer.setVisibility(View.VISIBLE);
                mvmodel.discoverPeers();
            } else {
                mBinding.screenContainer.setVisibility(View.GONE);
                mBinding.search.setVisibility(View.VISIBLE);
                mBinding.playerName.setVisibility(View.VISIBLE);
                mBinding.statusMsg.setText(R.string.no_one_connected_to);
            }
        });
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, String[] permissions,
                                           int[] grantResults) {
        if (requestCode == Utils.PERMISSIONS_REQUEST_CODE_ACCESS_COARSE_LOCATION
                && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
            mvmodel.requestPeers();
        }
    }

    @Override
    public void onResume() {
        super.onResume();
        mvmodel.onResume();
    }

    @Override
    public void onPause() {
        super.onPause();
        mvmodel.onPause();
    }
}

File: app/src/main/java/com/example/olesya/boardgames/ui/main/MenuFragment.kt
package com.example.olesya.boardgames.ui.main

import android.arch.lifecycle.ViewModelProviders
import android.content.pm.PackageManager
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.olesya.boardgames.R
import com.example.olesya.boardgames.databinding.FragmentMenuBinding
import com.example.olesya.boardgames.ui.main.MenuViewModel.PERMISSIONS_REQUEST_CODE_ACCESS_COARSE_LOCATION

class MenuFragment: Fragment() {

    private lateinit var mBinding: FragmentMenuBinding
    private lateinit var mViewModel: MenuViewModel

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        mBinding = DataBindingUtil.inflate(inflater, R.layout.fragment_menu, container, false)
        mViewModel = ViewModelProviders.of(this).get(MenuViewModel::class.java)
        initListeners()
        (activity as AppCompatActivity).supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        return mBinding.root
    }

    private fun initListeners() {
        mBinding.search.setOnClickListener(mViewModel.getOnSearchClickListener(activity, this))
        mBinding.start.setOnClickListener(mViewModel.getOnStartClickListener(mBinding.switcher,
                mBinding.playerName, mBinding.playerNum, mBinding.ptsNum))

        mBinding.switcher.setOnCheckedChangeListener { _, isChecked ->
            if (isChecked) {
                mBinding.statusMsg.text = "Server, waiting for connections"
                mBinding.search.visibility = View.GONE
                mBinding.playerName.visibility = View.GONE
                mBinding.screenContainer.visibility = View.VISIBLE
                mViewModel.discoverPeers()
            } else {
                mBinding.screenContainer.visibility = View.GONE
                mBinding.search.visibility = View.VISIBLE
                mBinding.playerName.visibility = View.VISIBLE
                mBinding.statusMsg.setText(R.string.no_one_connected_to)
            }
        }
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
        if (requestCode == PERMISSIONS_REQUEST_CODE_ACCESS_COARSE_LOCATION && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
            mViewModel.requestPeers()
        }
    }

    override fun onResume() {
        super.onResume()
        mViewModel.onResume()
    }

    override fun onPause() {
        super.onPause()
        mViewModel.onPause()
    }
}